docs/proposals/hpa-v2.md

```diff
@@ -0,0 +1,242 @@
+    TargetValue resource.Quantity `json:"targetValue"`
+}
+
+type HorizontalPodAutoscalerStatus struct {
+    // most recent generation observed by this autoscaler.
+    ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+    // last time the autoscaler scaled the number of pods;
+    // used by the autoscaler to control how often the number of pods is changed.
+    LastScaleTime *unversioned.Time `json:"lastScaleTime,omitempty"`
+
+    // the last observed number of replicas from the target object.
+    CurrentReplicas int32 `json:"currentReplicas"`
+    // the desired number of replicas as last computed by the autoscaler
+    DesiredReplicas int32 `json:"desiredReplicas"`
+
+    // current average CPU utilization percentage, set if a target CPU was requested
+    CurrentCPUUtilizationPercentage *int32 `json:"currentCPUUtilizationPercentage,omitempty"`
+    // the last read state of the metrics used by this autoscaler
+    CurrentMetrics []MetricStatus `json:"currentMetrics" protobuf:"bytes,5,rep,name=currentMetrics"`
+}
+
+// the status of a single metric
+type MetricStatus struct {
+    // the type of metric source
+    Type MetricSourceType `json:"type"`
+
+    // metric describing a single Kubernetes object
+    Object *ObjectMetricStatus `json:"object,omitemtpy"`
+    // metric describing pods in the scale target
+    Pods *PodsMetricStatus `json:"pods,omitemtpy"`
+}
+
+// a metric describing a Kubernetes object
+type ObjectMetricStatus struct {
+    // the described Kubernetes object
+    Target CrossVersionObjectReference `json:"target"`
+
+    // the name of the metric in question
+    MetricName string `json:"metricName"`
+    // the current value of the metric (as a quantity)
+    CurrentValue resource.Quantity `json:"targetValue"`
+}
+
+// metric describing pods in the scale target
+type PodsMetricStatus struct {
+    // the name of the metric in question
+    MetricName string `json:"metricName"`
+    // the current value of the metric (as a quantity)
+    CurrentValue resource.Quantity `json:"targetValue"`
+}
+```
+
+### Example ###
+
+In this example, we scale based on the `hits-per-second` value recorded as
+describing a service in our namespace, plus the CPU usage of the pods in
+the ReplicationController being autoscaled.
+
+```yaml
+kind: HorizontalPodAutoscaler
+apiVersion: autoscaling/v2alpha1
+spec:
+  scaleTargetRef:
+    kind: ReplicationController
+    name: WebFrontend
+  minReplicas: 2
+  maxReplicas: 10
+  targetCPUUtilizationPercentage: 80
+  metrics:
+  - object:
+      target:
+        kind: Service
+        name: Frontend
+      metricName: hits-per-second
+      targetValue: 1k
+```
+
+### Alternatives and Future Considerations ###
+
+Since the new design mirrors volume plugins (and similar APIs), it makes
+it relatively easy to introduce new fields in a backwards-compatible way:
+we simply introduce a new field in `MetricSpec` as a new "metric type".
+
+#### External ####
```

Comment:
I'd like to record that we considered it and rejected it for the moment.  I'll make it clear that it has unresolved questions and that we decided it was not necessary for the moment.